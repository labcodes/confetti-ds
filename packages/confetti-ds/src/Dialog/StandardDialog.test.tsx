/* eslint-disable no-alert */
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { StandardDialog } from "./index";

const args = {
  title: "Sample Message Dialog",
  icon: "StackStar" as const,
  content:
    "This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.",
  buttonProps: {
    text: "Required button",
    onClick: () => alert("Required button clicked"),
  },
  outlineButtonProps: {
    text: "Optional button",
    onClick: () => alert("Optional button clicked"),
  },
  isModal: false,
  isOpen: false,
  handleCLose: jest.fn(),
};

describe("StandardDialog", () => {
  afterEach(() => {
    /* reset html tag after each test case */
    document.getElementsByTagName("html")[0].innerHTML = "";
  });

  it("rendering with base props", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = <StandardDialog {...args} />;
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();
  });

  it("rendering with isOpen false", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = (
      <StandardDialog {...args} isModal isLarge isOpen={false} />
    );
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();

    expect(document.body.style.overflow).toBe("");
  });

  it("rendering with isModal false", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = (
      <StandardDialog {...args} isModal={false} isLarge isOpen />
    );
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });

  it("rendering with isLarge false", () => {
    expect(document.body.style.overflow).toBe("");

    const Component = (
      <StandardDialog {...args} isModal isOpen isLarge={false} />
    );
    const renderedComponent = renderer.create(Component);

    expect(renderedComponent.toJSON()).toMatchSnapshot();

    /* To clean up things */
    renderedComponent.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });

  it(`rendering if isOpen false sets body overflow to empty`, () => {
    expect(document.body.style.overflow).toBe("");

    const wrapper = mount(
      <StandardDialog
        {...args}
        isModal // true
        isOpen={false}
      />
    );

    expect(document.body.style.overflow).toBe("");

    expect(wrapper.find(".lab-dialog__title").exists()).toBeFalsy();
    expect(wrapper.find(".lab-dialog__body").exists()).toBeFalsy();

    wrapper.unmount();
  });
  it("checks if isOpen and isModal true sets body overflow to hidden", () => {
    expect(document.body.style.overflow).toBe("");

    const wrapper = mount(
      <StandardDialog
        {...args}
        isModal // true
        isOpen // true
      />
    );

    expect(document.body.style.overflow).toBe("hidden");

    expect(wrapper.find(".lab-dialog__title").text()).toBe(args.title);
    expect(wrapper.find(".lab-dialog__body").text()).toBe(args.content);

    wrapper.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });
  it("toggle isOpen", () => {
    expect(document.body.style.overflow).toBe("");

    const wrapper = mount(<StandardDialog {...args} isModal isOpen />);

    expect(document.body.style.overflow).toBe("hidden");

    expect(wrapper.find(".lab-dialog__title").text()).toBe(args.title);
    expect(wrapper.find(".lab-dialog__body").text()).toBe(args.content);

    wrapper.setProps({ isOpen: false });

    expect(document.body.style.overflow).toBe("unset");

    expect(wrapper.find(".lab-dialog__title").exists()).toBeFalsy();
    expect(wrapper.find(".lab-dialog__body").exists()).toBeFalsy();

    wrapper.unmount();

    expect(document.body.style.overflow).toBe("unset");
  });
});
