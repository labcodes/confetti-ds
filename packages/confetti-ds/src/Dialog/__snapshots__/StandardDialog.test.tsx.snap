// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardDialog rendering with base props 1`] = `null`;

exports[`StandardDialog rendering with isLarge false 1`] = `
[
  <div
    className="lab-dialog-overlay"
    onClick={[Function]}
    role="presentation"
  />,
  <div
    className="lab-dialog lab-dialog--standard
            "
  >
    <div
      aria-modal="true"
      className="lab-dialog__content"
      role="dialog"
    >
      <div
        className="lab-dialog__header"
      >
        <div
          className="lab-dialog__title"
        >
          Sample Message Dialog
        </div>
        <button
          className="lab-iconbtn lab-iconbtn--teal"
          onClick={[Function]}
        >
           
          <span
            className="lab-icon lab-icon--Teddy lab-icon--mineral-80 lab-iconbtn__icon"
          >
            <svg
              aria-hidden={true}
              className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 6.5a3.5 3.5 0 0 0 1.4 2.8c-.26.69-.4 1.43-.4 2.2 0 3.63 3.17 6.5 7 6.5s7-2.87 7-6.5c0-.77-.14-1.51-.4-2.2a3.5 3.5 0 1 0-5.31-4.19 7.54 7.54 0 0 0-2.58 0A3.5 3.5 0 0 0 2 6.5ZM5.5 4c.97 0 1.8.55 2.22 1.35a6.92 6.92 0 0 0-3.87 3.03A2.5 2.5 0 0 1 5.5 4Zm6.78 1.35a2.5 2.5 0 1 1 3.87 3.03 6.92 6.92 0 0 0-3.87-3.03ZM10 6c3.35 0 6 2.5 6 5.5a5.2 5.2 0 0 1-1.2 3.3 5.5 5.5 0 0 0-9.6 0A5.2 5.2 0 0 1 4 11.5C4 8.5 6.65 6 10 6Zm4.06 9.55A6.27 6.27 0 0 1 10 17c-1.57 0-3-.55-4.06-1.45a4.51 4.51 0 0 1 2.73-2.35 1.5 1.5 0 0 0 2.66 0c1.2.37 2.19 1.23 2.73 2.35Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
      <p
        className="lab-dialog__body"
      >
        This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.
      </p>
      <div
        className="lab-dialog__footer"
      >
        <button
          className="lab-btn lab-btn--outline lab-btn--normal lab-btn--teal"
          onClick={[Function]}
          tabIndex={3}
          type="button"
        >
          Optional button
        </button>
        <button
          className="lab-btn lab-btn--default lab-btn--normal lab-btn--teal"
          onClick={[Function]}
          tabIndex={1}
          type="button"
        >
          Required button
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`StandardDialog rendering with isModal false 1`] = `
<div
  className="lab-dialog lab-dialog--standard
             lab-dialog--large"
>
  <div
    aria-modal="true"
    className="lab-dialog__content"
    role="dialog"
  >
    <div
      className="lab-dialog__header"
    >
      <div
        className="lab-dialog__title"
      >
        Sample Message Dialog
      </div>
      <button
        className="lab-iconbtn lab-iconbtn--teal"
        onClick={[Function]}
      >
         
        <span
          className="lab-icon lab-icon--Teddy lab-icon--mineral-80 lab-iconbtn__icon"
        >
          <svg
            aria-hidden={true}
            className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 6.5a3.5 3.5 0 0 0 1.4 2.8c-.26.69-.4 1.43-.4 2.2 0 3.63 3.17 6.5 7 6.5s7-2.87 7-6.5c0-.77-.14-1.51-.4-2.2a3.5 3.5 0 1 0-5.31-4.19 7.54 7.54 0 0 0-2.58 0A3.5 3.5 0 0 0 2 6.5ZM5.5 4c.97 0 1.8.55 2.22 1.35a6.92 6.92 0 0 0-3.87 3.03A2.5 2.5 0 0 1 5.5 4Zm6.78 1.35a2.5 2.5 0 1 1 3.87 3.03 6.92 6.92 0 0 0-3.87-3.03ZM10 6c3.35 0 6 2.5 6 5.5a5.2 5.2 0 0 1-1.2 3.3 5.5 5.5 0 0 0-9.6 0A5.2 5.2 0 0 1 4 11.5C4 8.5 6.65 6 10 6Zm4.06 9.55A6.27 6.27 0 0 1 10 17c-1.57 0-3-.55-4.06-1.45a4.51 4.51 0 0 1 2.73-2.35 1.5 1.5 0 0 0 2.66 0c1.2.37 2.19 1.23 2.73 2.35Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
    <p
      className="lab-dialog__body"
    >
      This is the main content of a Message Dialog. You may want to put as much content as you want, as long as it's a string.
    </p>
    <div
      className="lab-dialog__footer"
    >
      <button
        className="lab-btn lab-btn--outline lab-btn--normal lab-btn--teal"
        onClick={[Function]}
        type="button"
      >
        Optional button
      </button>
      <button
        className="lab-btn lab-btn--default lab-btn--normal lab-btn--teal"
        onClick={[Function]}
        type="button"
      >
        Required button
      </button>
    </div>
  </div>
</div>
`;

exports[`StandardDialog rendering with isOpen false 1`] = `null`;
